lms/djangoapps/course_home_api/mixins.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework import serializers
+
+from opaque_keys.edx.keys import CourseKey
+from openedx.features.content_type_gating.models import ContentTypeGatingConfig
+from openedx.features.course_experience.utils import dates_banner_should_display
+
+
+class DatesBannerSerializerMixin(serializers.Serializer):
+    """
+    Serializer Mixin for displaying the dates banner.
+    Can be added to any serializer who's tab wants to display it.
+    """
+    dates_banner_info = serializers.SerializerMethodField()
+
+    def get_dates_banner_info(self, obj):
+        course_key_string = self.context['view'].kwargs.get('course_key_string')
+        info = {
+            'missed_deadlines': False,
+            'content_type_gating_enabled': False,
+        }
+        if course_key_string:
+            course_key = CourseKey.from_string(course_key_string)
+            request = self.context['request']
+            missed_deadlines, course_mode = dates_banner_should_display(course_key, request)
```

Comment:
If you aren't using `course_mode`, switch this to `missed_deadlines,  _ = ...`

Reply:
yea thats actually going to be removed anyway, was just waiting for cale to make his changes.